File: 0347_TopKFrequentElements/solution.go
// Time: 6ms (93.93%) | Memory : 5.92 (91.47%)
func topKFrequent(nums []int, k int) []int {
  m :=  make(map[int]int)    

  for x := range nums{
    m[nums[x]] += 1
  }
  
  keys  := make([]int, 0, len(m))
  for key := range m{
    keys = append(keys, key)
  }

  sort.SliceStable(keys, func(i,j int) bool{
    return m[keys[i]] > m[keys[j]]
  })
  return keys[:k]
}

File: 0347_TopKFrequentElements/solution.ts
// Time: 63ms (89.21%)  | Memory: 53.24 (88.40%)
function topKFrequent(nums: number[], k: number): number[] {
  if (nums.length === k) return nums;

  let map: Map<number, number> = new Map<number, number>();
  const len: number = nums.length;

  for (let i: number = 0; i < len; i++) {
    map.set(nums[i], (map.get(nums[i]) ?? 0) + 1);
  }
  const mapDesc: Map<number, number> = new Map(
    [...map.entries()].sort((a, b) => b[1] - a[1]),
  );
  return [...mapDesc.keys()].slice(0, k);
}
